database.class.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+    include_once 'databaseManager.class.php';
+
+    /**
+     * Database connection
+     */
+    class Database extends DatabaseManager
+    {
+        /**
+         * Constructor
+         */
+        public function __construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset)
+        {
+            parent::__construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset);
+        }
+
+        /**
+         * @return PDO
+         */
+        public function openConnection()
+        {
+            if ($this->isConnected() != 1)
+            {
+                try 
+                {                    
+                    $this->connection = new PDO($this->dbDriver.":host=".$this->host.";port=".$this->port.";charset=".$this->charset.";unix_socket=".$this->unixSocket.";dbname=".$this->dbName,$this->username,$this->password);
+                    echo "Successfull connection to database" . "\n";
+
+                    /* disable emulated prepared statements and use real prepared statements */
+                    $this->connection->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+                    $this->connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+                } catch (PDOException $e) 
+                {
+                    die($e->getMessage());
+                }  
+            }
+            return $this->connection;    
+        }
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        public function selectData($table, $conditions = array())
+        {
+            $sql = 'SELECT ';
+            $sql .= array_key_exists("select",$conditions)?$conditions['select']:'*';
+            $sql .= ' FROM '.$table;
+
+            if(array_key_exists("where",$conditions))
+            {
+                $sql .= ' WHERE ';
+                $i = 0;
+
+                foreach($conditions['where'] as $key => $value)
+                {
+                    $pre = ($i > 0)?' AND ':'';
+                    $sql .= $pre.$key." = '".$value."'";
+                    $i++;
+                }
+            }
+
+            if(array_key_exists("order_by",$conditions))
+            {
+                $sql .= ' ORDER BY '.$conditions['order_by']; 
+            }
+
+            if(array_key_exists("start",$conditions) && array_key_exists("limit",$conditions))
+            {
+                $sql .= ' LIMIT '.$conditions['start'].','.$conditions['limit']; 
+
+            } elseif(!array_key_exists("start",$conditions) && array_key_exists("limit",$conditions))
+            {
+                $sql .= ' LIMIT '.$conditions['limit']; 
```

Comment:
@lumakos FYI: `Limit...Offset` isn't a standard SQL Syntax. It might be supported MySQL, PostgreSQL, SQLite etc, but it doesn't by MS-SQL, Oracle, DB2 and a lot more.
So I guess pagination could be described by an interface.